### ISSUE TITLE
Add support for proposals on `work.ubq.fi` 

### ISSUE THREAD

**Issue by 0x4007** [ubiquity/.github#115]
Related https://github.com/ubiquity/work.ubq.fi/issues/54

There's a lot of unaccounted for/half completed proposals and it will be best to have a dedicated view to sort and browse. 

I recreated the task here because this involves work across several repositories. 

- The first thing that needs to be addressed is to upgrade the [backend](https://github.com/ubiquity/devpool-directory/issues). Perhaps we can make a new repository with a better name vs `devpool-directory` although it isn't clear to me what the name should be yet. Perhaps we can call it "organization issue aggregator" as sort of a general purpose name. It should list every issue in our organization, and include the price tag if there is a price. If not, then no price tag should be listed
- Next, on work.ubq.fi it should only filter for issues with price tags. 
- Then we can make a separate view that will filter for issues without price tags. It will allow us to keep the logic consolidated together. 
- Lastly we should figure something about syncing https://github.com/ubiquity/devpool-directory and https://github.com/ubiquity/devpool-directory-private I hope it can be rolled into this task but given that there is no details around how to implement, it might need to be broken off into a separate task.

**Comment by CaptainPrinz** [comment]
@Keyrxng before taking this task again, there's some questions I want to ask.
First what's the values for the environment variable in the .env file as modifying the file won't do any good if one doesnt know the values during development.

Second. In the backend, the code is complicated and one needs to study the where to actually extend the code.

I was hoping you could walk me through the file structure and all that before I start to work on nit.

Lastly in the future you may need to migrate to next.js or react and express to keep things structure as working on your current codebase is not an easy task.

Thanks .
@0x4007 @jordan-ae

**Comment by Keyrxng** [comment]
### Task Overview:

- **Overhaul of Core Infrastructure:**  
  - Update `devpool-directory` (backend) to maintain current functionality (state-sync, issues, labels, stats, etc.) in line with the spec.

- **Refactor `work.ubq.fi` UI:**  
  - Ensure a stable backend.
  - Implement filtering by price and add a view for unpriced (proposals).

- **Backend Repo Sync Solution:**  
  - Address the sync issue between the public `devpool-directory` and the private `private-devpool-directory`. The solution should allow the private fork to remain private while enabling easy syncing with the main repo.

---

> What's the value for the environment variables in the `.env` file? Modifying it won't help if the values are unknown during development.

This task involves at least three repos and for `devpool-directory`, you only need your `GITHUB_TOKEN`. `work.ubq.fi` requires your (recommended) Supabase instance key and URL or our [exposed](https://github.com/ubiquity/work.ubq.fi/blob/development/.github/workflows/cypress-testing.yml) key and url

> The backend code is complicated, and it's hard to know where to extend it.

Yes, this is a complex task, and that's reflected in the pricing. Understanding where and how to extend the code is part of the challenge, especially for your first few tasks. Start small.

> Can you walk me through the file structure before I start?

Unfortunately, that's not feasible for such a large task involving three codebases. At a high level:
- `work.ubq.fi` = frontend
- `devpool-directory` & `private-devpool-directory` = backend
  - Public Backend: Repo-issue based backend that aggregates tasks from the org
  - Private Backend: Same as public but restricted to those with permission

> You may need to migrate to Next.js or React with Express to improve structure. The current codebase is hard to work with.

Thanks for the suggestion. Some projects are moving to frameworks, but most will remain as-is for now.

---

@CaptainPrinz I appreciate your enthusiasm and willingness to tackle large tasks, though I’m concerned these are your first within the org. I recommend starting with smaller tasks to familiarize yourself with the codebases. 

If you're committed to this, please start opening PRs in the relevant repos so we can track progress. Since this is complex, can could maybe outline your approach first if you have thought that far yet?

**Comment by Keyrxng** [comment]
updating the above comment...

**Comment by rndquu** [comment]
So as a part of this issue we should:
1. Refactor https://github.com/ubiquity/devpool-directory to sync all available issues (right now it syncs only the ones with the "Price" label)
2. Refactor https://github.com/ubiquity/work.ubq.fi to display issues only with the "Price" label
3. Create a new "view" (i.e. github repository, perhaps with next.js, using css styles from https://github.com/ubiquity/work.ubq.fi) that will be able to display github issues from  https://github.com/ubiquity/devpool-directory based on env variables. Possible use cases:
  - show issues with the "Price" label
  - show issues without the "Price" label
 
 > Lastly we should figure something about syncing https://github.com/ubiquity/devpool-directory and https://github.com/ubiquity/devpool-directory-private I hope it can be rolled into this task but given that there is no details around how to implement, it might need to be broken off into a separate task.
 
 There's already issue for that https://github.com/ubiquity/devpool-directory-bounties/issues/25
 
> @rndquu does [this PR](https://github.com/ubiquity/devpool-directory/pull/1212) cover all aspects required for this larger task as well as the task in which it [links to](https://github.com/ubiquity/work.ubq.fi/issues/54)? - the issue timelines and PR threw me off at first, but below is correct yeah?

Current issue contradicts with https://github.com/ubiquity/work.ubq.fi/issues/54. If https://github.com/ubiquity/work.ubq.fi/issues/54 is not yet implemented then it may be closed as "not planned" in favor of https://github.com/ubiquity/.github/issues/115.

**Comment by Keyrxng** [comment]
> If https://github.com/ubiquity/work.ubq.fi/issues/54 is not yet implemented then it may be closed as "not planned" in favor of https://github.com/ubiquity/.github/issues/115.

Refactoring devpool [PR exists](https://github.com/ubiquity/devpool-directory/pull/1212) which is step 1 and required for step 2 I believe. If this is true then @xno-miner should be assigned to this task here and would be required to complete step 3 in order to be rewarded, is that right?

The sync issue will be treated as a separate task although the assignee should consider it if possible but is not a blocker for reward?

**Comment by rndquu** [comment]
> Refactoring devpool https://github.com/ubiquity/devpool-directory/pull/1212 which is step 1.

https://github.com/ubiquity/devpool-directory/pull/1212 introduces a feature of syncing either issues with the "Price" label either without it, not both. So it was meant to be forked and run in a separate repository. While the current issue describes that https://github.com/ubiquity/devpool-directory should sync all of the github issues. That's why https://github.com/ubiquity/devpool-directory/pull/1212 contradicts with the step 1 from [this](https://github.com/ubiquity/.github/issues/115#issuecomment-2284385715) comment.

> and required for step 2 I believe

As far as I understand step 2 does not depend on step 1. We can refactor `work.ubq.fi` to display github issues with the "Price" label right now.

> If this is true then @xno-miner should be assigned to this task here and would be required to complete step 3 in order to be rewarded, is that right?

Perhaps we could add https://github.com/ubiquity/work.ubq.fi/issues/54 reward to this issue and ask @xno-miner to start the current issue.

> The sync issue will be treated as a separate task although the assignee should consider it if possible but is not a blocker for reward?

Yes, it could be moved to a separate [task](https://github.com/ubiquity/devpool-directory-bounties/issues/25).

**Comment by CaptainPrinz** [comment]
/stop

@xno-miner can take care of it

**Comment by xno-miner** [comment]
/start

> 1. Refactor https://github.com/ubiquity/devpool-directory to sync all available issues (right now it syncs only the ones with the "Price" label)

Since doing it this way is going to require mostly different changes than those currently included in [ubiquity/devpool-directory#1212](https://github.com/ubiquity/devpool-directory/pull/1212), I'm going to close it and create a new PR. If anyone ever wanted to separate the logic again, they would just revive the old PR, because the code for it is already written.

> 2. Refactor https://github.com/ubiquity/work.ubq.fi to display issues only with the "Price" label
> 
> 3. Create a new "view" (i.e. github repository, perhaps with next.js, using css styles from https://github.com/ubiquity/work.ubq.fi) that will be able to display github issues from  https://github.com/ubiquity/devpool-directory based on env variables. Possible use cases:
> 
> 
>     * show issues with the "Price" label
> 
>     * show issues without the "Price" label

It would be best if the new backend I will make would get quickly merged or launched in a test repo, as it would make testing the frontend easier, but if it's not possible I'll figure something out.

**Comment by Keyrxng** [comment]
> It would be best if the new backend I will make would get quickly merged or launched in a test repo, as it would make testing the frontend easier, but if it's not possible I'll figure something out.

I'm sure you should be able to test everything locally between the two codebases without ubiquity hosting or merging to development, tag me here or on tg if you need help with anything.

**Comment by xno-miner** [comment]
I have done step 1 (https://github.com/ubiquity/devpool-directory/pull/1250, [xno-miner/devpool-directory/issues](https://github.com/xno-miner/devpool-directory/issues)) and step 2 (https://github.com/ubiquity/work.ubq.fi/pull/77).

I will start doing step 3 tommorow, but I think there's no need in creating an entirely new website. The current one works fine and there will be no problem in integrating the enviromental variables into it. I will first try to do it on this one and create a new one if any problems arise.

**Comment by xno-miner** [comment]
> I will start doing step 3 tommorow, but I think there's no need in creating an entirely new website. The current one works fine and there will be no problem in integrating the enviromental variables into it. I will first try to do it on this one and create a new one if any problems arise.

Done. https://github.com/ubiquity/work.ubq.fi/pull/77

**Comment by 0x4007** [comment]
We will split the task reward but also @gentlementlegen we need to fix payouts here.

**Comment by gentlementlegen** [comment]
@0x4007 I do not know the API keys, please update https://github.com/ubiquity-os-marketplace/conversation-rewards/settings/secrets/actions with the proper keys similar to our previous repo.

**Comment by 0x4007** [comment]
It was not changed in the last week?

<img width="570" alt="image" src="https://github.com/user-attachments/assets/ef2ae7c5-8fd2-4ec5-82cf-a99009350440">

**Comment by 0x4007** [comment]
Just unsuspended both apps on both ubiquity-os and ubiquity-os-marketplace orgs

**Comment by 0x4007** [comment]
> ```diff
> ! Failed to run comment evaluation. TypeError: Cannot read properties of undefined (reading 'wordValue')
> ```

@gentlementlegen

**Comment by gentlementlegen** [comment]
@0x4007 i believe this is fixed on develop not merged on main.

**Comment by 0x4007** [comment]
Good thing we test both!

**Comment by gentlementlegen** [comment]
Should work if you merge https://github.com/ubiquity-os-marketplace/text-conversation-rewards/pull/142

**Comment by xno-miner** [comment]
When should I be expecting to receive the payment? It has been some time since the changes have been merged

**Comment by 0x4007** [comment]
The last couple of times i tried to set up UUSD payments had blockers. Ill try again this week.

**Comment by gentlementlegen** [comment]
@0x4007 If you can assign the proper person and regenerate again, I do not know who you'd like to assign according to your comment https://github.com/ubiquity/.github/issues/115#issuecomment-2375499717

**Comment by 0x4007** [comment]
Your call I'm distracted with travels can only review in some weeks

**Comment by xno-miner** [comment]
Thank you
